scripts/tests/test_validate_unwanted_patterns.py

```diff
@@ -0,0 +1,77 @@
+import io
+
+# the "as" is just so I won't have to change this in many places later.
+import validate_string_concatenation as validate_unwanted_patterns
+
+
+class GoodBarePytestRaises:
+    def normal_pytest_raises(self):
+        return io.StringIO(
+            """
+with pytest.raises(ValueError, match="foo"):
+    pass
+"""
+        )
+
+    def pytest_raises_as_comment(self):
+        return io.StringIO(
+            """
+# with pytest.raises(ValueError, match="foo"):
+#     pass
+"""
+        )
+
+    def bare_pytest_raises_as_comment(self):
+        """
+        We do not care if the bare pytest raises is commented out.
+        """
+        return io.StringIO(
+            """
+# with pytest.raises(ValueError):
+#     pass
+"""
+        )
+
+
+class BadBarePytestRaises:
+    def normal_bare_pytest_raises(self):
+        return io.StringIO(
+            """
+with pytest.raises(ValueError):
+    raise ValueError("foo")
+"""
+        )
+
+
+class TestGoodBarePytestRaises:
```

Comment:
This is ok, but I think the class doesn't add value in this case, just have the tests as functions.

Reply:
I am planning to test "strings_to_concatenate" and "strings_with_wrong_placed_whitespace", this is just a sample file, because I'm not sure what I do here is good or not.

Should I keep the class then or not?

Reply:
Using classes to group the tests of each function sounds good. Up to you.